File: 26_RemoveDuplicatesFromSortedArray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  var i = 0;
  var tmp = 'a';
  while (i < nums.length) {
    if (tmp === nums[i]) {
      nums.splice(i, 1);
    } else {
      tmp = nums[i++];
    }
  }

  return nums;
};

File: 26.RemoveDuplicatesfromSortedArray.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length < 2) {
            return nums.length;
        }
        int result = 1;
        for (int i = 1; i < nums.length; i += 1) {
            if (nums[i - 1] != nums[i]) {
                nums[result] = nums[i];
                result += 1;
            }
        }
        return result;
    }
}